diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.elementType == ElementType.CLASS)
+                Assert.assertTrue(node.checkLength(IntRange(code.length, code.length)))
+        }
+    }
+
+    @Test
+    fun `test IdentifierName`(){
+        val code = """
+            class Test {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getIdentifierName() != null)
+                Assert.assertTrue(listOf("Test", "foo", "a", "Int").contains(node.getIdentifierName()!!.text))
+        }
+    }
+
+    @Test
+    fun `test getTypeParameterList`(){
+        val code = """
+            class Test(val idу: Int) {
+                private val id: Int = 10
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getTypeParameterList() != null) {
+                Assert.assertTrue(listOf("Test", "foo", "a", "Int").contains(node.getTypeParameterList()!!.text))
+            }
+        }
+    }
+
+    @Test
+    fun `test getAllIdentifierChildren`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        val list = mutableListOf<String>()
+        applyToCode(code){node ->
+            if (node.getAllIdentifierChildren().isNotEmpty())
+                node.getAllIdentifierChildren().forEach { list.add(it.text) }
+        }
+        Assert.assertTrue(list.containsAll(listOf("Test", "foo", "a", "Int")))
+    }
+
+    @Test
+    fun `test getAllChildrenWithType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            val list = mutableListOf<String>()
+            if (node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                node.getAllChildrenWithType(CLASS).forEach {
+                    list.add(it.text)
+                }
+                Assert.assertEquals(list, listOf(code))
+            }
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty()) {
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, node.getAllIdentifierChildren()[0].text)
```

Comment:
Generally, you shouldn't compare results of different methods, which you intend to test, before you have proved their validity by comparing to some static data. I see that in these tests you have common problem that lambda is executed for all nodes. To workaround this, you could create example which will be executed only once, e.g. without identifiers on deeper tree levels, or add check that `node.treeParent.elementType == FILE` or smth like this.